packages/@sanity/vision/src/components/VisionGui.tsx

```diff
@@ -587,6 +591,9 @@ export class VisionGui extends PureComponent<VisionGuiProps, VisionGuiState> {
 
     const queryStart = Date.now()
 
+    const ast = parse(query)
```

Comment:
This needs to be try/catched - if you have a syntax error or refer to unknown functions (say, `array::distinct` instead of `array::unique`, the tool throws an uncaught error toast and gets into an odd state where it appears to be executing the query, but is actually stalled.

Reply:
Should probably also catch the line below, in case that can throw too.